### PR TITLE
support for hardlink

### PR DESCRIPTION
**Pull request by vnashkev**
Methods 
         createHardlink( String linkname )  & 
         createHardlink( String linkname, boolean replaceIfExist ) 
have been adde to DiskEntry, 

Thank you in advance

### CODE REVIEW

File: src/main/java/com/hierynomus/smbj/share/DiskEntry.java [hunk 1]
```diff
@@ -106,6 +107,19 @@ public void rename(String newName, boolean replaceIfExist, long rootDirectory) t
         this.setFileInformation(renameInfo);
     }
 
+    public void createHardlink( final String linkname )
```
Comment (hierynomus) [inline src/main/java/com/hierynomus/smbj/share/DiskEntry.java]: Can you add JavaDocs and format correctly?

File: src/main/java/com/hierynomus/msfscc/fileinformation/FileLinkInformation.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hierynomus.msfscc.fileinformation;
+
+public class FileLinkInformation
```
Comment (hierynomus) [inline src/main/java/com/hierynomus/msfscc/fileinformation/FileLinkInformation.java]: Can you please use similar formatting as the rest of the project? I.e.:
- Egyptian braces
- no space after open- and before close-bracket
-

File: src/main/java/com/hierynomus/msfscc/fileinformation/FileInformationFactory.java [hunk 3]
```diff
@@ -285,6 +285,22 @@ public void write(FileRenameInformation info, Buffer outputBuffer) {
         };
         encoders.put(FileRenameInformation.class, renameCodec);
 
+        FileInformation.Encoder<FileLinkInformation> linkCodec = new FileInformation.Encoder<FileLinkInformation>()
```
Comment (hierynomus) [inline src/main/java/com/hierynomus/msfscc/fileinformation/FileInformationFactory.java]: Please reformat according to project style.

File: src/main/java/com/hierynomus/smbj/share/DiskEntry.java (hunk 1 above)
Comment (vnashkev) [reply src/main/java/com/hierynomus/smbj/share/DiskEntry.java]: Seems to be ready.
Could you please tell me why CI for oracle java 9 fails?

**Pull request merged**